pkg/router/template/router.go

```diff
@@ -203,38 +215,38 @@ func newTemplateRouter(cfg templateRouterCfg) (*templateRouter, error) {
 		peerEndpoints:            []Endpoint{},
 		bindPortsAfterSync:       cfg.bindPortsAfterSync,
 
+		commitReqTime: nil,
+		commitRunning: false,
+		stateChanged:  true,
+
+		reloadInterval:  cfg.reloadInterval,
+		reloadGap:       cfg.reloadGap,
+		reloadEventWait: cfg.reloadEventWait,
+
 		metricReload:      metricsReload,
 		metricWriteConfig: metricWriteConfig,
-
-		rateLimitedCommitFunction:    nil,
-		rateLimitedCommitStopChannel: make(chan struct{}),
 	}
 
-	numSeconds := int(cfg.reloadInterval.Seconds())
-	router.EnableRateLimiter(numSeconds, router.commitAndReload)
+	router.SetCommitFunc(func() error {
+		return router.commitAndReload()
+	})
+
+	glog.V(2).Infof("Template router will coalesce reloads within %s seconds of the last restart start time, within %s seconds of the last restart end time, and wait %s seconds after the first event", router.reloadInterval.String(), router.reloadGap.String(), router.reloadEventWait.String())
 
 	if err := router.writeDefaultCert(); err != nil {
 		return nil, err
 	}
+
 	glog.V(4).Infof("Reading persisted state")
 	if err := router.readState(); err != nil {
 		return nil, err
 	}
-	glog.V(4).Infof("Committing state")
-	// Bypass the rate limiter to ensure the first sync will be
-	// committed without delay.
-	router.commitAndReload()
-	return router, nil
-}
 
-func (r *templateRouter) EnableRateLimiter(interval int, handlerFunc ratelimiter.HandlerFunc) {
-	keyFunc := func(_ interface{}) (string, error) {
-		return "templaterouter", nil
-	}
+	// Do an immediate commit so that we can reply to health checks before the state syncs
```

Comment:
does this make the router reply that it is healthy prior to having synced? is that correct? couldn't that cause traffic to be directed to it prior to actually having all the routes set up?

Reply:
We already cover that behavior in the router... the haproxy template is passed a synced flag.  So it doesn't bind 80/443 until the sync happens, but it does bind the status port for health checks.

Reply:
That seems really odd. It reports healthy before it can serve traffic?